Topics autopartitioning in CDC YDBDOCS-1104 (#18034)
ydb-platform/ydb: ydb/docs/en/core/yql/reference/syntax/alter_table/changefeed.md

{% endif %}
* `VIRTUAL_TIMESTAMPS`: Enabling/disabling [virtual timestamps](../../../../concepts/cdc.md#virtual-timestamps). Disabled by default.
* `RETENTION_PERIOD`: [Record retention period](../../../../concepts/cdc.md#retention-period). The value type is `Interval` and the default value is 24 hours (`Interval('PT24H')`).

> * `TOPIC_AUTO_PARTITIONING`: [Topic autopartitioning mode](../../../../concepts/cdc.md#topic-partitions):
>     * `ENABLED` – An [autopartitioned topic](../../../../concepts/topic.md#autopartitioning) will be created for this changefeed. The number of partitions in such a topic increases automatically as the table update rate increases. Topic autopartitioning parameters [can be configured](../alter-topic.md#alter-topic).
>     * `DISABLED` - Topic without [autopartitioning](../../../../concepts/topic.md#autopartitioning) will be created for this changefeed. This is a default value.
> * `TOPIC_MIN_ACTIVE_PARTITIONS`: [The initial number of topic partitions](../../../../concepts/cdc.md#topic-partitions). By default, the initial number of topic partitions is equal to the number of table partitions. For autopartitioned topics, the number of partitions increases automatically as the table update rate increases.

Reviewer: ```suggestion
* `TOPIC_AUTO_PARTITIONING`: [Topic autopartitioning mode](../../../../concepts/cdc.md#topic-partitions):
    * `ENABLED` – An [autopartitioned topic](../../../../concepts/topic.md#autopartitioning) will be created for this changefeed. The number of partitions in such a topic increases automatically as the table update rate increases. Topic autopartitioning parameters [can be configured](../alter-topic.md#alter-topic).
    * `DISABLED` – A topic without [autopartitioning](../../../../concepts/topic.md#autopartitioning) will be created for this changefeed. This is the default value.
* `TOPIC_MIN_ACTIVE_PARTITIONS`: [The initial number of topic partitions](../../../../concepts/cdc.md#topic-partitions). By default, the initial number of topic partitions is equal to the number of table partitions. For autopartitioned topics, the number of partitions increases automatically as the table update rate increases.
```

{% endif %}
* `VIRTUAL_TIMESTAMPS`: Enabling/disabling [virtual timestamps](../../../../concepts/cdc.md#virtual-timestamps). Disabled by default.
* `RETENTION_PERIOD`: [Record retention period](../../../../concepts/cdc.md#retention-period). The value type is `Interval` and the default value is 24 hours (`Interval('PT24H')`).

> * `TOPIC_AUTO_PARTITIONING`: [Topic autopartitioning mode](../../../../concepts/cdc.md#topic-partitions):

Reviewer: By the way, why is all this only in the alter table article, but not in the create table?

>     * `ENABLED` - [autopartitioned topic](../../../../concepts/topic.md#autopartitioning) will be create for this changefeed. Number of partitions in such topic grows automatically as table update rate increases. Topic autopartitioning parameters [can be configured](../alter-topic.md#alter-topic).

Reviewer: Is there `DISABLED` too? In that case, we need to specify both and mark one as the default.

> * `TOPIC_AUTO_PARTITIONING`: [Topic autopartitioning mode](../../../../concepts/cdc.md#topic-partitions):
>     * `ENABLED` - [autopartitioned topic](../../../../concepts/topic.md#autopartitioning) will be create for this changefeed. Number of partitions in such topic grows automatically as table update rate increases. Topic autopartitioning parameters [can be configured](../alter-topic.md#alter-topic).
> * `TOPIC_MIN_ACTIVE_PARTITIONS`: [The initial number of topic partitions](../../../../concepts/cdc.md#topic-partitions). By default, the initial number of topic partitions is equal to the number of table partitions. For the autopartitioned topics number of partitions will grow automatically as table update rate increases.

Reviewer: ```suggestion
* `TOPIC_AUTO_PARTITIONING`: [Topic autopartitioning mode](../../../../concepts/cdc.md#topic-partitions):
    * `ENABLED` – An [autopartitioned topic](../../../../concepts/topic.md#autopartitioning) will be created for this changefeed. The number of partitions in such a topic increases automatically as the table update rate increases. Topic autopartitioning parameters [can be configured](../alter-topic.md#alter-topic).
* `TOPIC_MIN_ACTIVE_PARTITIONS`: [The initial number of topic partitions](../../../../concepts/cdc.md#topic-partitions). By default, the initial number of topic partitions is equal to the number of table partitions. For autopartitioned topics, the number of partitions increases automatically as the table update rate increases.
```

> * `TOPIC_MIN_ACTIVE_PARTITIONS`: [The initial number of topic partitions](../../../../concepts/cdc.md#topic-partitions). By default, the initial number of topic partitions is equal to the number of table partitions. For the autopartitioned topics number of partitions will grow automatically as table update rate increases.

Reviewer: Link to instructions on how to manually increase the number of partitions if the autopartitioning isn't enabled.

Author: It's impossible now. If there is no autopartitioning on changefeed, user can only delete changefeed and re-create it with another number of partitions. I'll add this disclaimer
